go-controller/pkg/util/subnet_annotations.go

```diff
@@ -194,3 +194,17 @@ func ParseNodeLocalNatIPAnnotation(node *kapi.Node) ([]net.IP, error) {
 
 	return nodeLocalNatIPs, nil
 }
+
+// GetNodeLocalIP returns the primary IP of the node as *net.IP
+func GetNodeLocalIP(node *kapi.Node) (*net.IP, error) {
```

Comment:
nit you could just return a net.IP here, since where you call it you end up dereferencing anyway. There's no real advantage to the pointer here because net.IP is just a slice of bytes, which are pass by reference anyway and all you care about is reading the value (not modifying it).

Reply:
the only reason I passed back a pointer is to save on the slice header. slice header is actually the part that gets deep copied and it's bigger than pointer width. Splitting hairs, but sometimes all the performance you can eke out matters. That being said, I am OK changing it to just `net.IP`